maro/cli/grass/executors/grass_on_premises_executor.py

```diff
@@ -0,0 +1,404 @@
+        )
+        copy_files_to_node(
+            local_path=f"{GlobalPaths.MARO_CLUSTERS}/{self.cluster_name}",
+            remote_dir=GlobalPaths.MARO_CLUSTERS,
+            admin_username=admin_username, node_ip_address=node_public_ip_address
+        )
+
+        # Remote init node
+        self.grass_executor.remote_init_node(
+            node_name=node_name,
+            node_ip_address=node_public_ip_address
+        )
+
+        # Get public key
+        public_key = self.grass_executor.remote_get_public_key(node_ip_address=node_public_ip_address)
+
+        # Save details
+        node_details["public_key"] = public_key
+        self.grass_executor.remote_set_node_details(
+            node_name=node_name,
+            node_details=node_details
+        )
+
+        # Update node status
+        # Since On-Premises machines don't need to shutdown, it will be set to start directly.
+        self.grass_executor.remote_update_node_status(
+            node_name=node_name,
+            action="start"
+        )
+
+        # Load images
+        self.grass_executor.remote_load_images(
+            node_name=node_name,
+            parallels=GlobalParams.PARALLELS,
+            node_ip_address=node_public_ip_address
+        )
+
+        # Load node agent service
+        self.grass_executor.remote_load_node_agent_service(
+            node_name=node_name,
+            node_ip_address=node_public_ip_address
+        )
+
+        logger.info_green(f"Node {node_name} has been initialized.")
+
+    def node_leave_cluster(self, node_name: str):
+        cluster_details = self.cluster_details
+        if "grass/on-premises" != cluster_details["mode"]:
+            logger.warning("This sub command only for On-Premises mode.")
+            return
+
+        nodes_details = self.grass_executor.remote_get_nodes_details()
+        if node_name not in nodes_details:
+            logger.warning(f"The specified node cannot be found in cluster {cluster_details['name']}.")
+            return
+
+        node_details = nodes_details[node_name]
+        # Update node status
+        self.grass_executor.remote_update_node_status(
+            node_name=node_name,
+            action='stop'
+        )
+        # Delete node record in redis.
+        self.grass_executor.remote_update_node_status(node_name, 'delete')
+
+        admin_username = cluster_details["user"]["admin_username"]
+        node_ip_address = node_details["public_ip_address"]
+        GrassOnPremisesExecutor.delete_user("", admin_username, node_ip_address)
+        logger.info_green(f"The node {node_name} has been left cluster {cluster_details['name']}.")
+
+    @staticmethod
+    def create_user(admin_username: str, maro_user: str, ip_address: str, pubkey: str) -> None:
+        if("" == admin_username):
+            print("Please input a user account that has permissions to create user:")
+            admin_username = input("> ")
```

Comment:
It' necessary. Because user sometimes uses password to connect linux server. I have already added some lines of code to retrieve super user account from YAML file.